core/vulkan/vk_memory_tracker_layer/cc/memory_tracker_layer_impl.cpp

```diff
@@ -1152,18 +1232,17 @@ void MemoryTracker::EmitAndClearAllStoredEvents() {
     }
     event_container_set->clear();
   }
-  rwl_devices.wunlock();
+  rwl_devices.runlock();
```

Comment:
These manual lock/unlock events make me somewhat uneasy.

You can submit, but maybe consider creating a small RAII struct to clean this up.

```
class scoped_read_locker {
    public:
      scoped_read_locker(rwlock* _lock): lock(_lock) { _lock.rlock(); }
     ~scoped_read_locker() { _lock.runlock(); }
     rwlock* lock
}

That way we can't accidently keep them around un-necessarily.